dask_cuda/disk_io.py

```diff
@@ -169,12 +169,16 @@ def disk_write(path: str, frames: Iterable, shared_filesystem: bool, gds=False)
         import kvikio
 
         with kvikio.CuFile(path, "w") as f:
-            for frame, length in zip(frames, frame_lengths):
-                f.pwrite(buf=frame, count=length, file_offset=0, buf_offset=0).get()
+            futs = []
+            file_offset = 0
+            for b, length in zip(frames, frame_lengths):
+                futs.append(f.pwrite(b, file_offset=file_offset))
+                file_offset += length
+            for each_fut in futs:
+                each_fut.get()
```

Comment:
Maybe pre-compute the file offsets and remove the `frame_lengths` all together?

```suggestion
            file_offsets = itertools.accumulate(map(nbytes, frames))
            futs = []
            for b, file_offset in zip(frames, file_offsets):
                futs.append(f.pwrite(b, file_offset=file_offset))
            for each_fut in futs:
                each_fut.get()
```

Reply:
Is there a reason to prefer this approach over what is in the PR?

Reply:
Not functionality wise but I think it is a bit more pythonic and concise.

Reply:
Would you be comfortable pushing those changes to this PrR?

Reply:
I have pushed changes that works for me locally.

Reply:
Thanks Mads! 🙏